### ISSUE TITLE
Sympa Accessibility Issue - 21 - Unable to select date from the calendar picker using keyboard

### ISSUE THREAD

**Issue by rm1-uiuc** [sympa-community/sympa#1751]
The University of Illinois at Urbana-Champaign, Technology Services Accessibility Testing team conducted an Accessibility Review of Sympa on September 25, 2023 and found 26 issues that need to be fixed to make the interface accessible for visual impaired users.

Version
-------
6.2.72

Installation method
-------------------
Source Package

Expected behavior
-----------------
User should be able to select a date in the calendar picker using a keyboard

Actual behavior
---------------
Unable to select date from the calendar picker using keyboard. When navigating through keyboard, the calendar picker popups when hitting enter but tabbing into the picker makes the picker disappear.

Steps to reproduce
------------------
Sympa Accessibility Issue - 21 was determined by manual functional testing performed by the user navigating with keyboard only or using assistive technology such as screen reader. See attached review for more info on specific location in interface.

Unable to select date from the calendar picker using keyboard. When navigating through keyboard, the calendar picker popups when hitting enter but tabbing into the picker makes the picker disappear.

[2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)

![image](https://github.com/sympa-community/sympa/assets/144282683/89dc597b-d90f-4d20-a94e-a77ad18609cf)
Fig 22 Keyboard accessibility issue on date picker.

Additional information
----------------------
Attached is the Accessibility Review conducted on Sept 25,2023
[Tech Services Review - Accessibility Evaluation Report for Sympa.pdf](https://github.com/sympa-community/sympa/files/13505476/Tech.Services.Review.-.Accessibility.Evaluation.Report.for.Sympa.pdf)

**Comment by ikedas** [comment]
This seems [a problem of jQuery UI Datapicker](https://stackoverflow.com/questions/12459562/jquery-ui-datepicker-configure-keyboard-shortcuts).  Its built-in keyboard shortcuts are not common.

Replacing with the other plug-in may fix this issue.

**Comment by ldidry** [comment]
Maybe we could simply use `<input type="date">` :thinking: 

- MDN: <https://developer.mozilla.org/fr/docs/Web/HTML/Element/input/date>
- Can I use: <https://caniuse.com/input-datetime>

**Comment by ikedas** [comment]
According to _Can I use_ information, some browsers have not support `date` type attribute (Additionally, IMO we should consider the possibility that earlier versions may survive in actual user environments).

And MDN  [notes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#handling_browser_support) that fallback to `text` type on unsupported browsers  will cause another problem: Ambiguity of date format.  In fact, when Sympa also have used `text` type for date entry, the ISO format `year-month-day` was very unpopular with some (typically North American) users: They often prefer to `month-day-year`.

Fortunately, several  accessibility-friendly alternatives to the jQuery UI DatePicker can be found, so why not select an appropriate one from among them?

**Comment by racke** [comment]
Definitely not (my 2 cents).

**Comment by ikedas** [comment]
I do not believe that much consideration needs to be given to browsers for PCs and smart devices. Users of legacy devices such as feature phones need to be considered, as their options are limited.

I noticed the Opera mini survey result in _Can I use_ is as of version 12.1. Can we confirm if recent versions support the date type attribute?

**Comment by ikedas** [comment]
With [Opera Mini for Android](https://www.opera.com/ja/browsers/opera-mini) 76.0.2254.69201, `date` type seems usable.

### [Semantics tests](https://wpt.live/html/semantics/forms/the-input-element/date.html) pass

![Screenshot_20231216-092259](https://github.com/sympa-community/sympa/assets/5743546/0fd53461-d16e-4497-8fac-9991fdda2540)

### [Rendering](https://wpt.live/html/rendering/widgets/input-date-content-size-ref.html) looks good

![Screenshot_20231216-091938](https://github.com/sympa-community/sympa/assets/5743546/b9e42def-b793-4185-81dd-904c75783d2c)

![Screenshot_20231216-091950](https://github.com/sympa-community/sympa/assets/5743546/eb47073e-afbd-47e0-af1e-e1268b81cca5)
